ambry-router/src/main/java/com/github/ambry/router/CompressionMetrics.java

```diff
@@ -0,0 +1,154 @@
+      compressError = registry.counter(MetricRegistry.name(ownerClass, algorithmName, "CompressError"));
+      compressRatioPercent = registry.histogram(MetricRegistry.name(ownerClass, algorithmName, "CompressRatioPercent"));
+      fullSizeCompressTimeInMicroseconds = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "FullSizeCompressTimeInMicroseconds"));
+      fullSizeCompressSpeedMBPerSec = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "FullSizeCompressSpeedMBPerSec"));
+      partialFullCompressTimeInMicroseconds = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "PartialFullCompressTimeInMicroseconds"));
+      partialFullCompressSpeedMBPerSec = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "PartialFullCompressSpeedMBPerSec"));
+
+      // Decompression metrics.
+      decompressRate = registry.meter(MetricRegistry.name(ownerClass, algorithmName, "DecompressRate"));
+      decompressError = registry.counter(MetricRegistry.name(ownerClass, algorithmName, "DecompressError"));
+      fullSizeDecompressTimeInMicroseconds = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "FullSizeDecompressTimeInMicroseconds"));
+      fullSizeDecompressSpeedMBPerSec = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "FullSizeDecompressSpeedMBPerSec"));
+      partialFullDecompressTimeInMicroseconds = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "PartialFullDecompressTimeInMicroseconds"));
+      partialFullDecompressSpeedMBPerSec = registry.histogram(MetricRegistry.name(ownerClass, algorithmName,
+          "PartialFullDecompressSpeedMBPerSec"));
+    }
+  }
+
+  // Group all compression related metrics under one owner "Compression".
+  // These metrics are per chunk, not per operation.
+  private final static Class<?> ownerClass = Compression.class;
+  private final MetricRegistry metricsRegistry;
+  private final Map<String, AlgorithmMetrics> algorithmToMetricsMap = new ConcurrentHashMap<>();
+
+  // Compression metrics, regardless of compression algorithm.
+  public final Meter compressSkipRate;      // The rate compression is skipped due to config settings.
+  public final Meter compressErrorRate;     // The overall compress failure rate, eg: compress() throws exception.
+  public final Meter compressAcceptRate;    // The rate compression is applied and accepted after filtering.
+  public final Counter compressReduceSizeBytes;  // For accepted, bytes reduce = OriginalSize - CompressedSize.
+
+  // Compression error metrics.  These metrics must be monitored and raise alarm if as soon as they occurred.
+  public final Counter compressError_BufferConversion;
+  public final Counter compressError_ConfigInvalidCompressorName;
+  public final Counter compressError_CompressFailed;
```

Comment:
Not a big fan of such naming pattern, care to change to camel case?

Reply:
Yes, I can change it to camel case.

Regarding categorizing the compression metrics.  Are we going use the metrics for analysis?  I presumption is we'll use it just for performance and stability purpose.  I don't think we should break it down to multiple categories unless there is a need for it.